src/socket/icmp.rs

```diff
@@ -297,7 +313,20 @@ impl<'a, 'b> IcmpSocket<'a, 'b> {
                         payload_len: repr.buffer_len(),
                         hop_limit:   hop_limit,
                     });
-                    emit((ip_repr, repr))
+                    emit((ip_repr, IcmpRepr::Ipv4(repr)))
+                },
+                IpAddress::Ipv6(ipv6_addr) => {
+                    let packet = Icmpv6Packet::new(&*packet_buf);
+                    let src_addr = Ipv6Address::default();
+                    let repr = Icmpv6Repr::parse(&src_addr.into(), &ipv6_addr.into(), &packet, &ChecksumCapabilities::default())?;
```

Comment:
Should we expect users to fill in the checksum? Or should we just parse it with `ChecksumCapabilities::ignored()`?

Reply:
We shouldn't expect the users to fill in checksum.

Reply:
@ProgVal you might update the corresponding line for ICMPv4 as well https://github.com/m-labs/smoltcp/pull/205/files#diff-7dff2f1b7b74bbfcd78f956578d349d1R308

Reply:
I don't really understand what `ChecksumCapabilities` mean (should the user of smoltcp verify them? or smoltcp itself? or the hardware?), but using `ChecksumCapabilities::ignored` fixed two of the issues, thanks!

Reply:
It is essentially used to determine if the parse/emit function should verify/fill the checksum. It is typically used in the context of a device. For example, a device that we can offload IPv4 checksumming on transmit to would have a `ChecksumCapabilities` structure with `ipv4` equal to [TX](https://github.com/m-labs/smoltcp/blob/master/src/phy/mod.rs#L127). However, in this case we don't want to evaluate the checksum so using `ChecksumCapabilities::ignored` makes that happen.